server/main_server/src/agent_main.cpp

```diff
@@ -0,0 +1,1326 @@
+        //
+
+        RsComm rsComm{};
+
+        // Originally set via an environment variable in initRsCommWithStartupPack in iRODS 4.2+.
+        rsComm.sock = _socket_fd;
+
+        irods::experimental::log::set_error_object(&rsComm.rError);
+        irods::at_scope_exit release_error_stack{[] { irods::experimental::log::set_error_object(nullptr); }};
+
+        rsComm.thread_ctx = static_cast<thread_context*>(std::malloc(sizeof(thread_context)));
+
+        bool require_cs_neg = false;
+        int status = initRsCommWithStartupPack(&rsComm, startupPack, require_cs_neg); // This version just uses the startupPack.
+
+        // manufacture a network object for comms
+        ret = irods::network_factory(&rsComm, net_obj);
+        if (!ret.ok()) {
+            log_agent::error(PASS(ret).result());
+        }
+
+        if (status < 0) {
+            log_agent::error("initRsCommWithStartupPack error: [{}]", status);
+            sendVersion(net_obj, status, 0, nullptr, 0);
+            cleanupAndExit(status);
+        }
+
+        // TODO This is initialized during agent factory startup. Why do we create a new one after forking the agent?
+        //irods::re_plugin_globals.reset(new irods::global_re_plugin_mgr);
+        // TODO This line invokes the start operation of all rule engine plugins. If a plugin opens a resource
+        // that cannot be shared (e.g. a database connection), then we have to come up with another way of
+        // achieving our goal of no config changes taking effect until a config reload/restart.
+        //
+        // All we want to do is make it so that configuration is static on server startup. Can a pre-startup
+        // entry point help with that? It would be run by the agent factory and only setup state for later use.
+        // Things that are considered a limited resource would NOT be handled here. This would make reloading of
+        // plugin configuration easier because the pre-startup logic is independent and well-defined. However, that's
+        // not something we need to worry about since the reload logic for the server simply forks new processes.
+        irods::re_plugin_globals->global_re_mgr.call_start_operations();
+
+        // TODO Cannot assume /var/lib/irods.
+        status = getRodsEnv(&rsComm.myEnv);
+
+        if (status < 0) {
+            log_agent::error("agentMain :: getRodsEnv failed");
+            sendVersion(net_obj, SYS_AGENT_INIT_ERR, 0, nullptr, 0);
+            cleanupAndExit(status);
+        }
+
+        std::string svc_role;
+        ret = get_catalog_service_role(svc_role);
+        if (!ret.ok()) {
+            log_agent::error(PASS(ret).result());
+            return 1;
+        }
+
+        // TODO Do we want to keep this?
+#if 0
+        if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+            if (std::strstr(rsComm.myEnv.rodsDebug, "CAT") != nullptr) {
+                chlDebug(rsComm.myEnv.rodsDebug);
+            }
+        }
+#endif
```

Comment:
This enables debug options for GenQuery1? I need to double check that. In any case, this isn't required for the server to do work.

Reply:
This controls debug / logging options for GenQuery1.

Open an issue for documenting this and handle it in a separate PR.

Reply:
Created #8009